Mathlib/Algebra/Algebra/Quasispectrum.lean

```diff
@@ -586,11 +586,4 @@ lemma quasispectrumRestricts_iff_spectrumRestricts_inr'
 
 theorem quasispectrumRestricts_iff_spectrumRestricts {R S A : Type*} [Semifield R] [Semifield S]
     [Ring A] [Algebra R S] [Algebra R A] [Algebra S A] {a : A} {f : S → R} :
-    QuasispectrumRestricts a f ↔ SpectrumRestricts a f := by
-  rw [quasispectrumRestricts_iff, spectrumRestricts_iff, quasispectrum_eq_spectrum_union_zero]
-  refine and_congr_left fun h ↦ ?_
-  refine ⟨(Set.RightInvOn.mono · Set.subset_union_left), fun h' x hx ↦ ?_⟩
-  simp only [Set.union_singleton, Set.mem_insert_iff] at hx
-  obtain (rfl | hx) := hx
-  · simpa using h 0
-  · exact h' hx
+    QuasispectrumRestricts a f ↔ SpectrumRestricts a f := by rfl
```

Comment:
For example: this *looks* like this was using many API lemmas before, and the `rfl` omits them all. Is this change kosher?

A similar question applies to most other `rfl` changes.

Reply:
[`SpectrumRestricts`](https://leanprover-community.github.io/mathlib4_docs/Mathlib/Algebra/Algebra/Quasispectrum.html#SpectrumRestricts) is just a type alias for `QuasispectrumRestricts`. Maybe this signals that having both at once just causes more confusion?

Reply:
ok, this rfl seems like it is not a defqe abuse. Perhaps @j-loreaux can take a look if further refactors around `QuasispectrumRestricts` are a good idea, but likely in a separate pr.

Reply:
The current proof was just a remnant from before I made it a type alias. They used to be different and this proof was required. The `rfl` here is a good change.